packages/swarm-components/src/Dropdown.jsx

```diff
@@ -0,0 +1,462 @@
+					{() => (
+						<Rect>
+							{({ rect: menuRect, ref: menuRef }) => (
+								<div
+									data-reach-menu
+									ref={menuRef}
+									style={getStyles(state.buttonRect, menuRect)}
+								>
+									<MenuListImpl {...props} ref={ref} />
+								</div>
+							)}
+						</Rect>
+					)}
+				</WindowSize>
+			</Portal>
+		)
+	);
+});
+
+MenuList.propTypes = {
+	children: node,
+};
+
+let isFocusableChildType = (child, types = []) => types.includes(child.type);
+let getFocusableMenuChildren = (children, types) => {
+	let focusable = [];
+	Children.forEach(children, child => {
+		if (isFocusableChildType(child, types)) focusable.push(child);
+	});
+	return focusable;
+};
+
+let MenuListImpl = React.forwardRef(
+	(
+		{
+			children,
+			onKeyDown,
+			onBlur,
+			focusableChildrenTypes = [MenuItem, MenuLink],
+			...rest
+		},
+		ref
+	) => {
+		const { state, setState, refs } = useContext(MenuContext);
+		let focusableChildren = getFocusableMenuChildren(
+			children,
+			focusableChildrenTypes
+		);
+		return (
+			<div
+				data-reach-menu-list
+				{...rest}
+				role="menu"
+				aria-labelledby={state.buttonId}
+				tabIndex="-1"
+				ref={node => {
+					refs.menu = node;
+					assignRef(ref, node);
+				}}
+				onBlur={event => {
+					if (!state.closingWithClick && !refs.menu.contains(event.relatedTarget)) {
+						setState({ ...state, ...close() });
+					}
+				}}
+				onKeyDown={wrapEvent(onKeyDown, event => {
+					if (event.key === 'Escape') {
+						setState({ ...state, ...close() });
+					} else if (event.key === 'ArrowDown') {
+						event.preventDefault(); // prevent window scroll
+						let nextIndex = state.selectionIndex + 1;
+						if (nextIndex !== focusableChildren.length) {
+							setState({ ...state, selectionIndex: nextIndex });
+						}
+					} else if (event.key === 'ArrowUp') {
+						event.preventDefault(); // prevent window scroll
+						let nextIndex = state.selectionIndex - 1;
+						if (nextIndex !== -1) {
+							setState({ ...state, selectionIndex: nextIndex });
+						}
+					} else if (event.key === 'Tab') {
+						event.preventDefault(); // prevent leaving
+					}
+				})}
+			>
+				{Children.map(children, (child, index) => {
+					if (isFocusableChildType(child, focusableChildrenTypes)) {
+						let focusIndex = focusableChildren.indexOf(child);
+
+						return React.cloneElement(child, {
+							setState,
+							state,
+							index: focusIndex,
+							_ref: node => (refs.items[focusIndex] = node),
+						});
+					}
+
+					return child;
+				})}
+			</div>
+		);
+	}
+);
+
+MenuListImpl.propTypes = {
+	refs: object,
+	state: object,
+	setState: func,
+	children: node,
+	onKeyDown: func,
+	onBlur: func,
+};
+
+let getStyles = (buttonRect, menuRect) => {
+	let haventMeasuredButtonYet = !buttonRect;
+	if (haventMeasuredButtonYet) {
+		return { opacity: 0 };
+	}
+
+	let haventMeasuredMenuYet = !menuRect;
+
+	let styles = {
+		left: `${buttonRect.left + window.pageXOffset}px`,
```

Comment:
if this is ever called on server render, it will crash because there is no `window`